### PR TITLE
:seedling: expose mqtt topics

### PR DESCRIPTION
**Pull request by skeeey**
## Related issue(s)

Fixes #

### CODE REVIEW

**Review by morvencao — COMMENTED** [review]
overall looks good.
grpc options should not need this, right?

**Comment by skeeey** [comment]
grpc may not this. this only for mqtt to set the share topic

**Comment by skeeey** [comment]
or we add an option for mqtt using share topic instead of setting topic？

File: cloudevents/generic/options/mqtt/options.go [hunk 1]
```diff
@@ -62,17 +66,31 @@ type MQTTConfig struct {
...
 	// KeepAlive is the keep alive time in seconds for MQTT clients, by default is 60s
 	KeepAlive *uint16 `json:"keepAlive,omitempty" yaml:"keepAlive,omitempty"`
+
 	// PubQoS is the QoS for publish, by default is 1
 	PubQoS *int `json:"pubQoS,omitempty" yaml:"pubQoS,omitempty"`
 	// SubQoS is the Qos for subscribe, by default is 1
 	SubQoS *int `json:"subQoS,omitempty" yaml:"subQoS,omitempty"`
+
+	// SpecTopic is a MQTT topic for resource spec, by default is sources/+/clusters/+/spec.
+	SpecTopic string `json:"specTopic,omitempty" yaml:"specTopic,omitempty"`
```
Comment (qiujian16) [inline cloudevents/generic/options/mqtt/options.go:76]: what about a topics struct and status/spect/statusResync in it. I feel like the struct will be reused by other messaging system

**Comment by skeeey** [comment]
/close

This pr will be opened on go-sdk repo

**Pull request closed without merging**